modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -1175,24 +1176,58 @@ public static byte[] byteBufferToByteArray(ByteBuffer buffer) {
         }
     }
 
+    private static CompletableFuture<Void> startAsync(Stream<? extends IgniteComponent> components) {
+        return allOf(components
+                .filter(Objects::nonNull)
+                .map(IgniteComponent::startAsync)
+                .toArray(CompletableFuture[]::new));
+    }
+
+    /**
+     * Asynchronously starts all ignite components.
+     *
+     * @param components Array of ignite components to start.
+     * @return CompletableFuture that will be completed when all components are started.
+     */
+    public static CompletableFuture<Void> startAsync(IgniteComponent... components) {
+        return startAsync(Stream.of(components));
+    }
+
+    /**
+     * Asynchronously starts all ignite components.
+     *
+     * @param components Collection of ignite components to start.
+     * @return CompletableFuture that will be completed when all components are started.
+     */
+    public static CompletableFuture<Void> startAsync(Collection<? extends IgniteComponent> components) {
+        return startAsync(components.stream());
+    }
+
+    private static CompletableFuture<Void> stopAsync(Stream<? extends IgniteComponent> components) {
```

Comment:
Please check my comment in modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerRecoveryTest.java